Sources/App/Controllers/ImageController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  ImageController.swift
+//  App
+//
+//  Created by Zach Eriksen on 10/19/19.
+//
+
+import Vapor
+import Imperial
+import Authentication
+
+struct ImageController: RouteCollection {
+    let imageFolder = "uploads/"
+    
+    func boot(router: Router) throws {
+        let authSessionRouter = router.grouped(User.authSessionsMiddleware())
+        
+        authSessionRouter.post("uploadImage", use: uploadImageHandler)
+        authSessionRouter.get("image", ImageUpload.parameter, use: getImageHandler)
+    }
+    
+    func uploadImageHandler(_ req: Request) throws -> Future<ImageUploadResponse> {
+        let user = try req.requireAuthenticated(User.self)
+        return try req.content
+            .decode(ImageUpload.self)
+            .flatMap { imageData in
+                let workPath = try req.make(DirectoryConfig.self).workDir
+                let name = try "\(user.requireID())-\(UUID().uuidString).jpg"
+                let path = workPath + self.imageFolder + name
```

Comment:
couldn't this just be created once on init?

Reply:
What part? The name var requires user authed. and UUID